### PR TITLE
Fix cob.js Crash

### PR DESCRIPTION
**Pull request by jpcunningh**
Under certain circumstances when NS has glucose calibration records, cob.js will crash due to avgDelta being undefined.

This PR updates the logic to avoid avgDelta the error causing the crash.

### CODE REVIEW

**Comment by jpcunningh** [comment]
For reference, the crash is:
```
/root/src/oref0/lib/determine-basal/cob.js:129
        avgDelta = avgDelta.toFixed(2);
                            ^

TypeError: Cannot read property 'toFixed' of undefined
    at detectCarbAbsorption (/root/src/oref0/lib/determine-basal/cob.js:129:29)
    at /root/src/oref0/lib/meal/total.js:68:39
    at Array.forEach (<anonymous>)
    at recentCarbs (/root/src/oref0/lib/meal/total.js:46:16)
    at generate (/root/src/oref0/lib/meal/index.js:20:19)
    at Object.<anonymous> (/root/src/oref0/bin/oref0-meal.js:97:23)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
```

File: lib/determine-basal/cob.js [hunk 1]
```diff
@@ -38,15 +44,8 @@ function detectCarbAbsorption(inputs) {
...
             bgTime = new Date(glucose_data[i].dateString);
         } else { console.error("Could not determine BG time"); }
-        if (bucketed_data[bucketed_data.length-1].display_time) {
-            lastbgTime = new Date(bucketed_data[bucketed_data.length-1].display_time.replace('T', ' '));
-        } else if (glucose_data[i-1].display_time) {
-            lastbgTime = new Date(glucose_data[i-1].display_time.replace('T', ' '));
-        } else if (glucose_data[i-1].dateString) {
-            lastbgTime = new Date(glucose_data[i-1].dateString);
-        } else { console.error("Could not determine last BG time"); }
-        if (! glucose_data[i].glucose || glucose_data[i].glucose < 39 || glucose_data[i-1].glucose < 39) {
```
Comment (scottleibrand) [inline lib/determine-basal/cob.js]: Why did you remove this instead of modifying it?

File: lib/determine-basal/cob.js [hunk 2]
```diff
@@ -66,11 +65,18 @@ function detectCarbAbsorption(inputs) {
                 continue;
             }
         }
+        if (bucketed_data[bucketed_data.length-1].display_time) {
+            lastbgTime = new Date(bucketed_data[bucketed_data.length-1].display_time.replace('T', ' '));
+        } else if ((lastbgi >= 0) && glucose_data[lastbgi].display_time) {
+            lastbgTime = new Date(glucose_data[lastbgi].display_time.replace('T', ' '));
+        } else if ((lastbgi >= 0) && glucose_data[lastbgi].dateString) {
+            lastbgTime = new Date(glucose_data[lastbgi].dateString);
+        } else { console.error("Could not determine last BG time"); }
```
Comment (scottleibrand) [inline lib/determine-basal/cob.js]: Why did you move these?

Comment (scottleibrand) [reply lib/determine-basal/cob.js]: Never mind, I see the commit comment now: “Move lastbgTime determination past code that skips old data to avoid unmeaningful errors in the log.”

File: lib/determine-basal/cob.js (hunk 1 above)
Comment (jpcunningh) [reply lib/determine-basal/cob.js]: Lines 41-47 were moved lower and lines 48 and 49 were modified to remove the check against the glucose_data[i-1].glucose value and are now lines 47 and 48.

The [i-1] check is no longer required due to how lastbgi is managed.
